snapcraft/_options.py

```diff
@@ -224,33 +235,58 @@ def __init__(self, use_geoip=False, parallel_builds=True,
         self._set_machine(target_deb_arch)
         self.__debug = debug
 
-    def is_host_compatible_with_base(self, base):
-        codename = None
+    def is_host_compatible_with_base(self, base: str) -> bool:
+        """Determines if the host is compatible with the GLIBC of the base.
+
+        The system should warn early on when building using a host that does
+        not match the intended base, this mechanism here enables additional
+        logic when that is ignored to determine built projects will actually
+        run.
+
+        :param str base: the base core snap to search for linker.
+        :returns: True if there are no GLIBC incompatibilities with the chosen
+                  build host, else it returns False.
+        :rtype: bool
+        """
+        codename = None  # type: str
         with suppress(errors.OsReleaseCodenameError):
             codename = OsRelease().version_codename()
             logger.debug('Running on {!r}'.format(codename))
 
-        # TODO do this more generically
-        if base == 'core18':
-            return codename in ('trusty', 'xenial', 'bionic')
-        else:
-            return codename in ('trusty', 'xenial')
+        build_host_for_base = _HOST_CODENAME_FOR_BASE.get(
+            base)  # type: str
+        compatible_hosts = _HOST_COMPATIBILITY.get(
+            build_host_for_base, [])  # type: List[str]
+        return codename in compatible_hosts
 
-    def get_core_dynamic_linker(self, base):
+    def get_core_dynamic_linker(self, base: str, expand: bool=True) -> str:
         """Returns the dynamic linker used for the targeted core.
-        If not found realpath for `/lib/ld-linux.so.2` is returned.
-        However if core is not installed None will be returned.
+
+        :param str base: the base core snap to search for linker.
+        :param bool expand: expand the linker to the actual linker if True,
```

Comment:
We have sort of similar symlink following logic elsewhere, and the parameters we tend to use (and the ones Python itself tends to use) tend to say something about the symlinks themselves. For example, `follow_symlinks`. Would it make sense to name this parameter `follow_symlinks` to indicate what exactly is happening?

Reply:
about `follow_symlinks`, the semantics are a bit different, as is the logic, not sure I prefer it. Feel free to counter back :slightly_smiling_face: